docs/usage/Templating.md

```diff
@@ -0,0 +1,245 @@
+
+For each of these purposes, a list of executions can be specified. Every execution can use a different template engine. The results of all specified executions with the same purpose will be merged. 
+
+For detailed information of blueprints see the [Blueprint Docs](./Blueprints.md).
+
+- [Templating](#templating)
+    - [Template Execution](#template-execution)
+    - [State Handling](#state-handling)
+    - [Template Engines](#template-engines)
+      - [Go Template](#go-template)
+      - [Spiff](#spiff)
+
+### Template Execution
+
+Landscaper uses templating for various purposes, most prominently in the blueprint (e.g. for deployitem and export generation). The dedicated section in the respective manifests is always a list of template execution configurations. Each execution is defined by a set of attributes:
+- **`name`** *string*
+  The _name_ is used for providing error messages during the templating execution. It is also used as an identifier for the [state](#state-handling) of the execution.
+
+- **`type`** *string*
+  The _type_ specifies which template engine should be used. Currently supported types are [`GoTemplate`](#go-template) and [`Spiff`](#spiff).
+
+- **`file`** *string* [optional]
+  If this property is set, the template is read from the specified file of the blueprint file structure. Exactly one of `file` and `template` has to be specified.
+
+- **`template`** *template* [optional]
+  If this property is set, the template is read from the given inline data, according to the specification of the specified template engine type. Exactly one of `file` and `template` has to be specified.
+
+The the rendered output of the templating must always be a YAML document. The document is expected to be a map. The structure is the same, independent of which template engine is used. The expected result is always read from a dedicated key, depending on the execution (e.g. `deployItems` for deployitem executions).
+
+**Example**
+```yaml
+deployExecutions:
+- name: my-spiff-template
+  type: Spiff
+  template:
+    deployItems:
+    - name: my-first-deploy-item
+      type: landscaper.gardener.cloud/mock
+      config: ...
+- name: my-go-template
+  type: GoTemplate
+  template: |
+    deployitems:
+    - name: my-second-deploy-item
+      type: landscaper.gardener.cloud/mock
+      config: ...
+```
+
+### Filesystem
+
+The blueprint's filesystem structure is accessible for the template engines as root file system.
+
+**Example**
+- Filesystem
+  ```
+  my-blueprint
+  ├── data
+  │   ├── template
+  │   └── config
+  └── blueprint.yaml
+  ```
+- Execution snippet from blueprint.yaml
+  ```yaml
+  - name: my-go-template
+    type: GoTemplate
+    file: "data/template"
+  ```
+- Template file
+  ```
+  deployitems:
+    - name: my-second-deploy-item
+      type: landscaper.gardener.cloud/mock
+      config:
+  {{ include "data/config" . | indent 6 }}
+  ```
```

Comment:
Why do we only describe the access to the blueprint file system. We have other access possibilities in the template like the imports, component descriptor - or do I misunderstand here something?

Reply:
perhaps just a link would be sufficient

Reply:
I added a paragraph with a link pointing to the blueprint rendering docs.

Reply:
👍